documentation/openapi-authoring-automated-guidelines.md

```diff
@@ -49,6 +49,7 @@ We request OpenAPI(Swagger) spec authoring be assigned to engineers who have an
 | M2062	| PutResponseResourceValidation | The 200 response model for an ARM PUT operation must have x-ms-azure-resource extension set to true in its hierarchy. Operation: '{0}' Model: '{1}'. | Error |
 | M3027	| TrackedResourceListByResourceGroup | The tracked resource, '{0}', must have a list by resource group operation. | Error |
 | M3027	| TrackedResourceListBySubscription | The tracked resource, '{0}', must have a list by subscriptions operation. | Error |
```

Comment:
noticed saw some inconstancy between rule and details: `TrackedResourceListBySubscription ` is id with `M3027 ` but in details it's `R3028` ... With this PR of separate we need to make sure to have consistency on what we output and what we document. Because, @veronicagg's incoming chnage on having links on VSCode and JsonOutput would need that to work good.